sentry_sdk/_types.py

```diff
@@ -20,11 +20,21 @@
     from typing import Mapping
     from typing import NotRequired
     from typing import Optional
+    from typing import Sequence
     from typing import Tuple
     from typing import Type
     from typing import Union
     from typing_extensions import Literal, TypedDict
 
+    SdkInfo = TypedDict(
+        "SdkInfo",
+        {
+            "name": str,
+            "version": str,
+            "packages": Sequence[Any],
+        },
+    )
+
```

Comment:
A few nitpicks here, in order from most to least important:
  - We can likely make the `packages` type stricter, `Sequence[Mapping[str, str]]` seems like it will work (you will need to import `Mapping` from `collections.abc`).
  - I think the class-based `TypedDict` syntax might be cleaner here.
  - Given that SDK is usually capitalized, I would name this type as `SDKInfo` rather than `SdkInfo`.

```suggestion
    class SDKInfo(TypedDict):
        name: str
        version: str
        packages: Sequence[Mapping[str, str]]

```

Reply:
Can we actually use the class based `TypedDict` in the SDK? I thought we used this for compatibility reasons and all other instances of `TypedDict` in the SDK uses this syntax.

Reply:
We use it already:

https://github.com/getsentry/sentry-python/blob/2c1e31c5390310ae696108aa135c055452600f43/sentry_sdk/tracing.py#L43

Perhaps what you said was true with Sentry SDK 1.x, but we can use class-based syntax now.